crates/rattler_conda_types/src/repo_data/mod.rs

```diff
@@ -176,15 +177,75 @@ impl RepoData {
     }
 }
 
+/// An error that can occur when parsing a platform from a string.
+#[derive(Debug, Error, Clone, Eq, PartialEq)]
+pub enum ConvertSubdirError {
+    #[error("platform: {platform}, arch: {arch} is not a known combination")]
+    NoKnownCombination {
+        /// The platform string that could not be parsed.
+        platform: String,
+        /// The architecture.
+        arch: String,
+    },
+    #[error("platform key is empty in index.json")]
+    PlatformEmpty,
+    #[error("arch key is empty in index.json")]
+    ArchEmpty,
+}
+
+/// Determine the subdir based on result taken from the prefix.dev
+/// database
+/// These were the combinations that have been found in the database.
+/// and have been represented in the function.
+///
+/// # Why can we not use Platform::FromStr?
+///
+/// We cannot use the Platform FromStr directly because x86 and x86_64
+/// are different architecture strings. Also some combinations have been removed,
+/// because they have not been found.
+fn determine_subdir(
+    platform: Option<String>,
+    arch: Option<String>,
+) -> Result<String, ConvertSubdirError> {
+    let platform = platform.ok_or(ConvertSubdirError::PlatformEmpty)?;
+    let arch = arch.ok_or(ConvertSubdirError::ArchEmpty)?;
+    let canonical = format!("{platform}-{arch}");
```

Comment:
Instead of first converting to string I would just match on (platform, arch).

Reply:
Changed it!